File: Code/Quickstarts/cs/UIProcessQuickstarts_AdvancedHostDemo/Client/StartMeUp.cs
//===============================================================================
// Microsoft User Interface Process Application Block for .NET Quick Start
// http://msdn.microsoft.com/library/en-us/dnbda/html/uip.asp
//
// StartMeUp.cs
//
// This file contains the implementations of the StartMeUp class.
//
//
//===============================================================================
// Copyright (C) 2000-2001 Microsoft Corporation
// All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR
// FITNESS FOR A PARTICULAR PURPOSE.
//==============================================================================

using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using Microsoft.ApplicationBlocks.UIProcess;

namespace Client
{
	/// <summary>
	/// Summary description for StartMeUp.
	/// </summary>
	public class StartMeUp : System.Windows.Forms.Form, IShutdownUIP
	{
		private ITask _startingTask;
		private System.Windows.Forms.Button button1;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public StartMeUp()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.button1 = new System.Windows.Forms.Button();
			this.SuspendLayout();
			//
			// button1
			//
			this.button1.Location = new System.Drawing.Point(64, 104);
			this.button1.Name = "button1";
			this.button1.Size = new System.Drawing.Size(168, 24);
			this.button1.TabIndex = 0;
			this.button1.Text = "Start Hosted Control Demo";
			this.button1.Click += new System.EventHandler(this.button1_Click);
			//
			// StartMeUp
			//
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(292, 266);
			this.Controls.Add(this.button1);
			this.Name = "StartMeUp";
			this.Text = "StartMeUp";
			this.Closing += new System.ComponentModel.CancelEventHandler(this.StartMeUp_Closing);
			this.Load += new System.EventHandler(this.StartMeUp_Load);
			this.ResumeLayout(false);

		}
		#endregion

		private void button1_Click(object sender, System.EventArgs e)
		{
			UIPManager.RegisterShutdown(this);
			this.Visible = false;
			_startingTask = new TestTask();
			UIPManager.StartUserControlsTask("demo",_startingTask);
		}

		private void StartMeUp_Load(object sender, System.EventArgs e)
		{

		}

		private void StartMeUp_Closing(object sender, System.ComponentModel.CancelEventArgs e)
		{

		}

		public ITask StartingTask
		{
			get
			{
				return _startingTask;
			}
		}
		#region IShutdownUIP Members

		public void Shutdown()
		{
			this.Visible = true;
		}

		#endregion
	}
}

File: Code/Quickstarts/vb/UIProcessQuickstarts_AdvancedHostDemo/Client/StartMeUp.vb
 '===============================================================================
' Microsoft User Interface Process Application Block for .NET Quick Start
' http://msdn.microsoft.com/library/en-us/dnbda/html/uip.asp
'
' StartMeUp.vb
'
' This file contains the implementations of the StartMeUp class.
'
' 
'===============================================================================
' Copyright (C) 2000-2001 Microsoft Corporation
' All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR
' FITNESS FOR A PARTICULAR PURPOSE.
'==============================================================================

Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports Microsoft.ApplicationBlocks.UIProcess

Namespace Client

    ' <summary>
    ' Summary description for StartMeUp.
    ' </summary>
    Public Class StartMeUp
        Inherits System.Windows.Forms.Form
		Implements IShutdownUIP
        Private _startingTask As ITask
        Private WithEvents button1 As System.Windows.Forms.Button

        ' <summary>
        ' Required designer variable.
        ' </summary>
        Private components As System.ComponentModel.Container = Nothing

        Public Sub New()
            '
            ' Required for Windows Form Designer support
            '
            InitializeComponent()
            '
            ' TODO: Add any constructor code after InitializeComponent call
            '
        End Sub

        ' <summary>
        ' Clean up any resources being used.
        ' </summary>
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub

#Region "Windows Form Designer generated code"

        ' <summary>
        ' Required method for Designer support - do not modify
        ' the contents of this method with the code editor.
        ' </summary>
        Private Sub InitializeComponent()
            Me.button1 = New System.Windows.Forms.Button
            Me.SuspendLayout()
            ' 
            ' button1
            ' 
            Me.button1.Location = New System.Drawing.Point(64, 104)
            Me.button1.Name = "button1"
            Me.button1.Size = New System.Drawing.Size(168, 24)
            Me.button1.TabIndex = 0
            Me.button1.Text = "Start Hosted Control Demo"
            ' 
            ' StartMeUp
            ' 
            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.ClientSize = New System.Drawing.Size(292, 266)
            Me.Controls.Add(button1)
            Me.Name = "StartMeUp"
            Me.Text = "StartMeUp"
            Me.ResumeLayout(False)
        End Sub

#End Region

        Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button1.Click
            UIPManager.RegisterShutdown(Me)
            Me.Visible = False
            _startingTask = New TestTask
			UIPManager.StartUserControlsTask("demo", _startingTask)
        End Sub


        Private Sub StartMeUp_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
        End Sub

        Private Sub StartMeUp_Closing(ByVal sender As Object, ByVal e As System.ComponentModel.CancelEventArgs) Handles MyBase.Closing
        End Sub

        Public ReadOnly Property StartingTask() As ITask
            Get
                Return _startingTask
            End Get
        End Property

#Region "IShutdownUIP Members"

        Public Sub Shutdown() Implements IShutdownUIP.Shutdown
            Me.Visible = True
        End Sub

#End Region

    End Class

End Namespace